### PR TITLE
Longmetric id support

### PR DESCRIPTION
**Pull request by ashncoder**

### CODE REVIEW

**Comment by ashncoder** [comment]
@chrbayer84 build is green. We can try this out tomorrow on shard14

File: carbonj.service/src/main/java/com/demandware/carbonj/service/db/points/DataPointRecord.java [hunk 1]
```diff
@@ -31,13 +31,14 @@ public static double toValue(byte[] valueBytes)
...
-    public static int toMetricId(byte[] keyBytes)
+    public static Long toMetricId(byte[] keyBytes)
     {
-        return Ints.fromBytes( keyBytes[0], keyBytes[1], keyBytes[2], keyBytes[3] );
+        return Longs.fromBytes( keyBytes[0], keyBytes[1], keyBytes[2], keyBytes[3], keyBytes[4],
+                keyBytes[5], keyBytes[6], keyBytes[7]);
     }
 
     public static int toTimestamp(byte[] keyBytes)
     {
```
Comment (chrbayer84) [inline carbonj.service/src/main/java/com/demandware/carbonj/service/db/points/DataPointRecord.java:41]: Great find! I was looking for this... The values were cut off and looked like not all bytes were considered..

File: carbonj.service/src/main/java/com/demandware/carbonj/service/db/index/NameRecordSerializer.java [hunk 2]
```diff
@@ -54,11 +60,67 @@ public String key( byte[] keyBytes )
...
+    public NameRecord toIndexEntry( String key, byte[] valueBytes )
     {
+        if(longId) {
+            return toIndexEntryWithLongId(key, valueBytes);
+        }
+        return toIndexEntryWithIntId(key, valueBytes);
+    }
+
+    @Override
+    public byte[] keyBytes(String key)
```
Comment (chrbayer84) [inline carbonj.service/src/main/java/com/demandware/carbonj/service/db/index/NameRecordSerializer.java:72]: Is this actually overriding anything?

Comment (chrbayer84) [reply carbonj.service/src/main/java/com/demandware/carbonj/service/db/index/NameRecordSerializer.java:72]: Ah, nevermind, this was just caught up in the greedy diff algorithm..

File: carbonj.service/src/main/resources/application.yml [hunk 3]
```diff
@@ -11,7 +11,8 @@ spring:
     active: dev
 metrics:
   store:
-    enabled: false
+    enabled: true
```
Comment (chrbayer84) [inline carbonj.service/src/main/resources/application.yml:14]: Let's leave the default to be disabled here.

File: carbonj.service/src/main/java/com/demandware/carbonj/service/db/points/DataPointRecord.java [hunk 4]
```diff
@@ -10,13 +10,23 @@
 import com.google.common.io.ByteStreams;
 import com.google.common.primitives.Ints;
 import com.google.common.primitives.Longs;
+import com.google.common.primitives.UnsignedBytes;
+import org.hibernate.validator.internal.metadata.aggregated.rule.ReturnValueMayOnlyBeMarkedOnceAsCascadedPerHierarchyLine;
```
[the remark this answers is not in the document]
Comment (ashncoder) [reply carbonj.service/src/main/java/com/demandware/carbonj/service/db/points/DataPointRecord.java]: Donw

File: carbonj.service/src/main/resources/application.yml (hunk 3 above)
Comment (ashncoder) [reply carbonj.service/src/main/resources/application.yml]: Done

File: carbonj.service/src/main/java/com/demandware/carbonj/service/admin/CarbonjAdmin.java [hunk 5]
```diff
@@ -164,6 +169,10 @@ public void dumpNames( @RequestParam( value = "startId", required = false, defau
         }
     }
 
+    private long getMaxId() {
+        return longId ? Long.MAX_VALUE : Integer.MAX_VALUE;
```
Comment (chrbayer84) [inline carbonj.service/src/main/java/com/demandware/carbonj/service/admin/CarbonjAdmin.java:173]: Good catch!

**Review by chrbayer84 — APPROVED** [review]

**Pull request merged**